pkg/blockstorage/awsefs/awsefs.go

```diff
@@ -92,7 +90,7 @@ func NewEFSProvider(ctx context.Context, config map[string]string) (blockstorage
 
 	efsVault, ok := config[awsconfig.ConfigEFSVaultName]
 	if !ok || efsVault == "" {
-		efsVault = defaultK10BackupVaultName
+		return nil, errors.New("EFS vault name is empty")
```

Comment:
Now that you have changed an assumption, I would expect that this would break some unit test in this package ? If not, then that means tests are missing . If so please add them.

Reply:
In k10, `efs_kube_test` will need to be updated. There is no unit test for `NewEFSProvider` in `kanister` however. Let me look into how best test it since the function actually validates creds with AWS